### ISSUE TITLE
Inconsistency in naming of preproccessing object in plpModel - causes applyTidy not to run in some cases on test

### ISSUE THREAD

**Issue by egillax** [OHDSI/PatientLevelPrediction#310]
@jreps , In the latest develop you added checks like this in ```predict.R```

```R
if(!is.null(plpModel$preprocessing$tidyCovariates))
```
both  for ```applyTidyCovariates``` and ```applyFeatureengineering```

The issues is that this object ```plpModel$preprocessing``` is inconsistently named for different models. For [Cyclops](https://github.com/OHDSI/PatientLevelPrediction/blob/develop/R/CyclopsModels.R#L177) and [KNN](https://github.com/OHDSI/PatientLevelPrediction/blob/develop/R/KNN.R#L125) it's called ```preprocess``` while for [RClassifier](https://github.com/OHDSI/PatientLevelPrediction/blob/develop/R/RClassifier.R#L62) and [Scikit-learn](https://github.com/OHDSI/PatientLevelPrediction/blob/develop/R/SklearnClassifier.R#L94) it's called ```preprocessing``` 

I also notice that some of the ```plpModel``` objects, like ```preprocessing``` and ```modelDesign``` don't depend on anything calculated in the model fit function. I think it would be good to refactor those out of there, maybe into ```fitPlp```, so those objects that can are only defined in one place as part of the standard pipeline. It would be easier to maintain. 

Btw I noticed this because my model had 0.5 AUC on the test set, because it didn't do normalization for the test set. I see that we don't have tests to check if tidy or the feature engineering is applied to the test set and I wanted to add those. But I'm not sure what would be the best way, any ideas?

**Comment by solis9753** [comment]
In addition:
[Line 178](https://github.com/OHDSI/PatientLevelPrediction/blob/62460414f146436793e1f46efa142078d3d86f15/R/CyclopsModels.R#L178) of the `fitCyclopsModel()` will be always be empty since we save the featureEngineer settings as an attribute to the plpData object, not the covariateData object. See for example [this](https://github.com/OHDSI/PatientLevelPrediction/blob/62460414f146436793e1f46efa142078d3d86f15/R/FeatureEngineering.R#L148)

**Comment by jreps** [comment]
So that just needs to be 
featureEngineering = attr(trainData, "metaData")$featureEngineering 
?

I have a better idea that a warning, I'll add metaData to the prediction to track whether FE and preprocessing were done. Then, in the tests we can see whether the FE metaData is F if FE is not specified in the model and the preprocessing is F if the preprocessing is not specified in the model but T otherwise.  The warning may be annoying as FE is generally not used, but each run of PLP will cause a warning.

**Comment by jreps** [comment]
fixed in the main branch now
